lib/resource_tools/json.rb

```diff
@@ -15,6 +54,12 @@ def json(&block)
     private :json
   end
 
+  # @todo Consider switch to ActiveSupport::HashWithIndifferentAccess
+  # We've remove our Hashie::Mash dependency in the event warehouse, where
+  # ActiveSupport::HashWithIndifferentAccess was a near drop-in replacement.
+  # Here however we have additional use of converting method calls into key
+  # lookups. It should be possible to replace these with explicit method
+  # definitions, but this fell outside the scope of a simple refactor.
```

Comment:
I took a crack at this, but started getting a little large for what was intended to be a cheeky refactor.